apps/nft-service/src/nft/ipfs.ts

```diff
@@ -3,19 +3,21 @@
 // https://github.com/TypeStrong/ts-node#help-my-types-are-missing
 import pinataSDK from '@pinata/sdk';
 import { log } from '@thecointech/logging';
-import { MetadataJson } from '@thecointech/nft-contract';
 import { validateImage, validateJson } from './validate';
-
+import type { MetadataJson } from '@thecointech/nft-contract';
 import { Readable } from 'stream';
 
 const pinata = pinataSDK(process.env.PINATA_API_KEY!, process.env.PINATA_API_SECRET!);
 
 //
 // Upload buffer to IPFS, return upload hash
-export async function upload(file: Buffer) {
+export async function upload(file: Buffer, filename: string) {
   log.trace(`Uploading ${file.byteLength} bytes`);
-
   const stream = Readable.from(file);
+
+  // Fixes upload issue: https://github.com/PinataCloud/Pinata-SDK/issues/28
+  // ¡¡ THE HACK !!
+  (stream as any).path = filename;
```

Comment:
The filename is supposed to be irrelevant, as IPFS addresses are content-dependent.  This API is only a few months old though.  Yay for the bleeding edge.